### PR TITLE
Django 2.0 Fixes.

### PR DESCRIPTION
**Pull request by paulmonk**
- Added `on_delete` values for the models and migrations where needed.
- Updated Tox config.
- Added a Pipfile.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -59,13 +62,17 @@ def get_version(package):
         'Framework :: Django',
```
Comment (lpomfrey) [inline setup.py:62]: Drop 1.9 and 1.10 here, and lose Python 3.3.

File: test_project/urls.py [hunk 2]
```diff
@@ -1,17 +1,26 @@
-from django.conf.urls import patterns, include, url
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+import django
+from pkg_resources import parse_version
+
+if parse_version(django.__version__) >= parse_version('2.0.0'):
```
Comment (lpomfrey) [inline test_project/urls.py]: Use a `try ... except ImportError` not an `if`, and don't use `path()` use `url()` instead.

File: test_project/urls.py [hunk 3]
```diff
@@ -1,17 +1,26 @@
...
+    from django.conf.urls import include, url
 
 # Uncomment the next two lines to enable the admin:
 # from django.contrib import admin
 # admin.autodiscover()
 
-urlpatterns = patterns('',
+urlpatterns = [
     # Examples:
-    # url(r'^$', 'test_project.views.home', name='home'),
```
Comment (lpomfrey) [inline test_project/urls.py]: Leave these `url()`s as they were, just omit the `patterns()` in favour of a plain list.

File: tox.ini [hunk 4]
```diff
@@ -1,38 +1,24 @@
...
+    {py34,py35,py36}-2.0.x
+    {py35,py36}-master
+skip_missing_interpreters=True
 
 [testenv]
-PYTHONPATH = {toxinidir}:{toxinidir}/debreach
-commands = python setup.py test
-install_command = pip install {opts} {packages}
 deps =
-    django
```
Comment (lpomfrey) [inline tox.ini:9]: Don't use github tarballs, install versions from pypi (except for master).

**Comment by paulmonk** [comment]
I've added commits based on your feedback.

File: tox.ini [hunk 5]
```diff
@@ -1,38 +1,23 @@
...
+# from this directory.
 [tox]
-envlist = py27, py27-dj18, py33-dj18, py34, py34-dj18, py35, py35-dj18, pypy, pypy-dj18
+envlist =
+    {py27,py34,py35}-1.8.x
+    {py27,py34,py35}-1.9.x
+    {py27,py34,py35}-1.10.x
+    {py27,py34,py35,py36}-1.11.x
+    {py34,py35,py36}-2.0.x
+    {py35,py36}-master
```
Comment (lpomfrey) [inline tox.ini]: master will need to have an associated github tarball in deps. Or just don't test against master.

**Comment by lpomfrey** [comment]
Can you update the .travis.yml file as well.

File: tox.ini (hunk 5 above)
Comment (paulmonk) [reply tox.ini]: Sorry, that was accidental.

Comment (paulmonk) [reply tox.ini]: I've removed the reference to master now.

**Comment by paulmonk** [comment]
Sure. I wasn't sure if you wanted to do that.

**Pull request merged**